Tasks/InstallSSHKeyV0/installsshkey-util.ts

```diff
@@ -104,15 +103,32 @@ export class SshToolRunner {
     constructor() {
     }
 
+    private isWindows(): boolean {
+        return !!os.type().match(/^Win/);
+    }
+
     private getExecutable(executable: string):string {
-        let isWindows: RegExpMatchArray = os.type().match(/^Win/);
-        if (isWindows && this.baseDir) {
+        if (this.isWindows() && this.baseDir) {
             executable = path.join(this.baseDir, path.join(this.sshGitExternalsDir, executable));
             executable += '.exe';
         }
         return executable;
     }
 
+    private restrictPermissionsToFile(fileLocation: string): void {
+        if (this.isWindows()) {
+            const userName: string = os.userInfo().username;
+            tl.execSync('icacls', [fileLocation, '/inheritance:r']);
+            tl.execSync('icacls', [fileLocation, '/grant:r', `${userName}:(F)`]);
```

Comment:
Do we need these extra modifications? Shouldn't the chmodSync call take care of it?

Reply:
Calling 'chmodSync' seems to be not enough because of differences between Windows and Unix permissions systems particularly permissions are inherited. The key file must be available only for the current user to be used with ssh-keygen on Windows agent otherwise 'Permissions are too open' error is thrown.  All other options I tried appeared to be ineffective.

Reply:
Hm, interesting - that makes sense, thanks for clarifying!